metabox/metabox/core/runner.py

```diff
@@ -40,24 +40,33 @@ def __init__(self, args):
         logger.add(
             sys.stdout,
             format=self._formatter,
-            level=args.log_level)
+            level=args.scenario_log_level)
         logger.level("TRACE", color="<w><dim>")
         logger.level("DEBUG", color="<w><dim>")
         # session config
-        if not args.config.exists():
+        if not args.config or not args.config.exists():
             raise SystemExit('Config file not found!')
         else:
             self.config = read_config(args.config)
             self.config = guess_source_uri(self.config)
             validate_config(self.config)
         # effective set of machine configs required by scenarios
         self.combo = set()
-        self.scn_variants = None
+        self.scn_variants = []
         self.machine_provider = None
         self.failed = False
-        self.tags = set(self.args.tags or [])
-        self.exclude_tags = set(self.args.exclude_tags or [])
-        self.hold_on_fail = self.args.hold_on_fail
+        try:
+            self.tags = set(self.args.tags or [])
+        except AttributeError:
+            self.tags = []
+        try:
+            self.exclude_tags = set(self.args.exclude_tags or [])
+        except AttributeError:
+            self.exclude_tags = []
+        try:
+            self.hold_on_fail = self.args.hold_on_fail
+        except AttributeError:
+            self.hold_on_fail = False
```

Comment:
Again, I'm not sure what conditions are required to raise an `AttributeError` exception here. Either `--hold-on-fail` has been set in the CLI and `self.args.hold_on_fail` equals `True`, either it's not been set in the CLI and `self.args.hold_on_fail` equals False...